### PR TITLE
Add support for EDIF files and update Zynq UART support

### PR DESCRIPTION
**Pull request by sgherbst**
## Summary
This PR adds support for EDIF files (vendor-neutral gate-level netlists), which makes it easier to work with IPs from tools outside of the Xilinx ecosystem.  It also updates the existing Zynq UART code and includes testing of the UART capabilities in the regression test.

## UART updates
1. Allowed the user to provide their own firmware files, including ``main.c`` (this is useful to fill the performance/flexibility gap between synthesizable test features and Python-driven test features).
2. Added firmware generator that produces ``get_`` and ``set_`` C functions for signals registered in the sim control YAML file.
3. Added automation for ``xsct`` for building the ``*.elf`` file and downloading it to the FPGA.
4. Added control bit ``valid`` to the control interface for Zynq CPU outputs, which allows the user to set up the register address and data without updating the register outputs.

### CODE REVIEW

File: anasymod/templates/xsct_build.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+class TemplXSCTBuild:
+    def __init__(self, sdk_path, proc_name='ps7_cortexa9_0',
```
Comment (sgherbst) [inline anasymod/templates/xsct_build.py:2]: note that proc_name is zynq-specific

File: anasymod/templates/xsct_program.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
+class TemplXSCTProgram:
+    def __init__(self, sdk_path, cpu_filter='"ARM*#0"', hw_name='hw',
```
Comment (sgherbst) [inline anasymod/templates/xsct_program.py:2]: cpu_filter is zynq-specific

File: anasymod/templates/xsct_program.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+fpga "{sdk_path / hw_name / top_name}.bit"
+
+# make the debugger aware of the memory map
+# TODO: is this needed?
+puts "Setting up the debugger..."
+loadhw "{sdk_path / top_name}.hdf"
+
+# initialize the processor
+puts "Initializing the processor..."
+source "{sdk_path / hw_name / 'ps7_init'}.tcl"
```
Comment (sgherbst) [inline anasymod/templates/xsct_program.py:33]: these three lines are zynq-specific

File: anasymod/templates/zynq_gpio.py [hunk 4]
```diff
@@ -0,0 +1,144 @@
+from anasymod.sim_ctrl.datatypes import DigitalSignal
```
Comment (sgherbst) [inline anasymod/templates/zynq_gpio.py:1]: this whole file is zynq-specific

**Review by gabrielrutsch — APPROVED** [review]
Looing fine as discussed

**Pull request merged**